apps/src/applab/AppLabView.jsx

```diff
@@ -84,16 +178,19 @@ var AppLabView = React.createClass({
             id="visualizationResizeBar"
             className="fa fa-ellipsis-v" />
         {this.props.instructionsInTopPane && <TopInstructions
-          isEmbedView={this.props.isEmbedView}
-          puzzleNumber={this.props.puzzleNumber}
-          stageTotal={this.props.stageTotal}
-          height={topPaneHeight}
-          markdown={this.props.instructionsMarkdown}
-          collapsed={this.props.instructionsCollapsed}
-          onToggleCollapsed={this.props.toggleInstructionsCollapsed}
-          onChangeHeight={this.props.setInstructionsHeight}/>
+            ref="topInstructions"
+            isEmbedView={this.props.isEmbedView}
+            puzzleNumber={this.props.puzzleNumber}
+            stageTotal={this.props.stageTotal}
+            height={topPaneHeight}
+            maxHeight={this.props.instructionsMaxHeight}
+            markdown={this.props.instructionsMarkdown}
+            collapsed={this.props.instructionsCollapsed}
+            onToggleCollapsed={this.props.toggleInstructionsCollapsed}
+            onChangeHeight={this.props.setInstructionsHeight}/>
```

Comment:
Probably in part because these grew over time. It might make sense to just connect this to the store. I'll think about it some more. If I do so, I'll probably do that as a separate PR.